craft_providers/bases/ubuntu.py

```diff
@@ -177,98 +177,6 @@ def _ensure_os_compatible(self, executor: Executor) -> None:
                 )
             )
 
-    def _update_apt_sources(self, executor: Executor, codename: str) -> None:
-        """Update the codename in the apt source config files.
-
-        :param codename: New codename to use in apt source config files (i.e. 'lunar')
-        """
-        apt_source = "/etc/apt/sources.list"
-        apt_source_dir = "/etc/apt/sources.list.d/"
-        cloud_config = "/etc/cloud/cloud.cfg"
-
-        # get the current ubuntu codename
-        os_release = self._get_os_release(executor=executor)
-        version_codename = os_release.get("VERSION_CODENAME")
-        logger.debug("Updating apt sources from %r to %r.", version_codename, codename)
-
-        # replace all occurrences of the codename in the `sources.list` file
-        sed_command = ["sed", "-i", f"s/{version_codename}/{codename}/g"]
-        try:
-            self._execute_run(
-                [*sed_command, apt_source],
-                executor=executor,
-                timeout=self._timeout_simple,
-            )
-        except subprocess.CalledProcessError as error:
-            raise BaseConfigurationError(
-                brief=f"Failed to update {apt_source!r}.",
-                details=details_from_called_process_error(error),
-            ) from error
-
-        # if cloud-init and cloud.cfg isn't present, then raise an error
-        try:
-            self._execute_run(
-                ["test", "-s", cloud_config],
-                executor=executor,
-                timeout=self._timeout_simple,
-            )
-        except subprocess.CalledProcessError as error:
-            raise BaseConfigurationError(
-                brief=(
-                    f"Could not update {cloud_config!r} because it is empty or "
-                    "does not exist."
-                ),
-                details=details_from_called_process_error(error),
-            ) from error
-
-        # update cloud.cfg to prevent the sources.list file from being reset
```

Comment:
so I understand that the motivation for this PR is that the noble images moved the sources file, but how come we don't have to do all of this anymore? As opposed to updating the code to use the new file?

Reply:
We can update source files in the ubuntu daily devel image, but we run into [this bug](https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1728616) when we try to update source files in the buildd daily devel image.

Why does that bug only affect buildd daily images and not ubuntu daily images?  No idea!

Last week, @sergiusens said we could stop updating the source files in devel images as long as we expire the base instances every 14 days (which is [already implemented](https://github.com/canonical/craft-providers/blob/19c6a2baf2fd1cc8b5706158c6ed5ebc79fe9363/craft_providers/lxd/lxd_provider.py#L149))

Reply:
So for regular images, we don't need to update the source files?

Reply:
Correct, this was only for [`devel` images](https://github.com/canonical/craft-providers/pull/504/files#diff-c042ac17cabf0d18e3301f2f93aede60dd3a57f6952f3ee1bac6b48a3d62fa03L297)

Reply:
ah I see; the change that I skipped over was the removal of the `if self.alias == BuilddBaseAlias.DEVEL:` code a bit further down on the file.